salt/orch/prepare-product-migration.sls

```diff
@@ -0,0 +1,104 @@
+{#- Get a list of nodes seem to be down or unresponsive #}
+{#- This sends a "are you still there?" message to all #}
+{#- the nodes and wait for a response, so it takes some time. #}
+
+{%- set default_batch = salt['pillar.get']('default_batch', 5) %}
+{#- tx_update_migration_available needs to be last, as it is still used as a selector til reboot #}
+{%- set migration_grains = ['tx_update_migration_notes', 'tx_update_migration_newversion', 'migration_in_progress', 'tx_update_migration_available'] %}
+
+{%- set nodes_down = salt.saltutil.runner('manage.down') %}
+{%- if nodes_down|length >= 1 %}
```

Comment:
Minor comment: I think you can replace `if nodes_down|length >= 1` by `if nodes_down`...

Reply:
are you sure that this will work? havben't tested it

Reply:
I guess it should work, but don't worry...